dependency_list_export.go

```diff
@@ -0,0 +1,107 @@
+package gitlab
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+)
+
+type DependencyListExportService struct {
+	client *Client
+}
+
+// createDependencyListExportOptions represents the available CreateDependencyListExport()
+// options.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/dependency_list_export.html#create-a-pipeline-level-dependency-list-export
+type createDependencyListExportOptions struct {
+	ExportType string `url:"export_type" json:"export_type"`
+}
+
+// DependencyListExport represents a request for a GitLab project's dependency list.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/dependency_list_export.html#create-a-pipeline-level-dependency-list-export
+type DependencyListExport struct {
+	ID          int    `json:"id"`
+	HasFinished bool   `json:"has_finished"`
+	Self        string `json:"self"`
+	Download    string `json:"download"`
+}
+
+// CreateDependencyListExport creates a new CycloneDX JSON export for all the project dependencies
+// detected in a pipeline.
+//
+// If an authenticated user does not have permission to read_dependency, this request returns a 403
+// Forbidden status code.
+//
+// SBOM exports can be only accessed by the export’s author.
+//
+// GitLab docs:
+// https://docs.gitlab.com/ee/api/dependency_list_export.html#create-a-pipeline-level-dependency-list-export
+func (s *DependencyListExportService) CreateDependencyListExport(id int, options ...RequestOptionFunc) (*DependencyListExport, *Response, error) {
+	// POST /pipelines/:id/dependency_list_exports
+	createExportPath := fmt.Sprintf("pipelines/%d/dependency_list_exports", id)
+
+	opt := &createDependencyListExportOptions{
+		ExportType: *Ptr("sbom"),
```

Comment:
That's interesting 🤔 I wonder if we should already expose the `CreateDependencyListExportOptions` as a parameter so that if there will ever be another `export_type` than `"sbom"` it wouldn't introduce a breaking change.

To guarantee the same user-friendliness we could just set the `ExportType` to "sbom" in case either the `CreateDependencyListExportOptions` is `nil` or the `ExportType` field is `nil`. 

WDYT?

Reply:
I agree. I'll change it.

Reply:
I defaulted the `CreateDependencyListExportOptions` to contain "ExportType: sbom" if it's `nil`, but for the `ExportType`, since it cannot be `nil` I decided to let it send whatever string value the user sets to the API, because in that case the user has to have explicitly set it to `""` .